rust/connlib/clients/android/src/lib.rs

```diff
@@ -305,6 +305,42 @@ impl Callbacks for CallbackHandler {
             None
         })
     }
+
```

Comment:
@thomaseizinger For context, the reason this is required is because Android's DNS API has no Split DNS functionality, which means we need to intercept *all* queries and pass the non-Firezone ones back to the System resolvers.

All other platforms _do_ have native SplitDNS, so for those platforms we only intercept all queries when `dns_fallback_strategy` is set to `upstream_resolver`.